cytominer_eval/operations/util.py

```diff
@@ -131,3 +138,102 @@ def get_grit_entry(df: pd.DataFrame, col: str) -> str:
         len(entries.unique()) == 1
     ), "grit is calculated for each perturbation independently"
     return str(list(entries)[0])
+
+
+class MahalanobisEstimator:
+    """
+    Store location and dispersion estimators of the
+    empirical distribution of data provided in an
+    array and allow computation of statistical
+    distances
+    """
+
+    def __init__(self, arr: Union[pd.DataFrame, np.ndarray]):
+        self.sigma = EmpiricalCovariance().fit(arr)
+
+    def mahalanobis(self, X: Union[pd.DataFrame, np.ndarray]) -> np.ndarray:
+        """
+        Compute the mahalanobis distance between
+        the empirical distribution described by
+        this object and points in an array `X`
+        """
+        return self.sigma.mahalanobis(X)
+
+
+def calculate_mahalanobis(pert_df: pd.DataFrame, control_df: pd.DataFrame) -> pd.Series:
+    """
+    Usage: Designed to be called within a pandas.DataFrame().groupby().apply()
+    """
+    assert len(control_df) > 1, "Error! No control perturbations found."
+
+    # Get dispersion and center estimators for the control perturbations
+    control_estimators = MahalanobisEstimator(control_df)
+
+    # Distance between mean of perturbation and control
+    maha = control_estimators.mahalanobis(np.array(np.mean(pert_df, 0)).reshape(1, -1))[
+        0
+    ]
+    return maha
+
+
+def default_mp_value_parameters():
+    """
+    Set the different default parameters used for mp-values.
+
+    Output:
+    A dictionary with the following keys:
+    rescale_pca - whether the PCA should be scaled by variance explained
+    nb_permutations - how many permutations to do to get empirical p-value
+    """
+    params = {"rescale_pca": True, "nb_permutations": 100}
+    return params
+
+
+def calculate_mp_value(
+    pert_df: pd.DataFrame, control_df: pd.DataFrame, params={}
```

Comment:
```suggestion
    pert_df: pd.DataFrame, control_df: pd.DataFrame, params: dict = {},
```

type hints. Maybe needs a run of black after accepting this commit